serving/docs/lmi_new/deployment_guide/configurations.md

```diff
@@ -1,6 +1,189 @@
+We recommend using the quick start configurations as a starting point if you have decided on a particular backend.
+The only change required to the starter configurations is specifying `option.model_id` to point to your model artifacts.
+
+We will now cover the components of a minimal starting configuration. This minimal configuration will look like:
+
+```
+# use standard python engine, or mpi aware python engine
+engine=<Python|MPI>
+# where the model artifacts are stored
+option.model_id=<hf_hub_model_id|s3_uri>
+# which inference library to use
+option.rolling_batch=<auto|vllm|lmi-dist|deepspeed|tensorrtllm>
+# how many gpus to shard the model across with tensor parallelism
+option.tensor_parallelism=<max|number between 1 and number of gpus available>
+```
+
+There are additional configurations that can be specified.
+We will cover the common configurations (across backends) in [LMI Common Configurations](#lmi-common-configurations)
+
+## Model Artifact Configuration
+
+If you are deploying a model hosted on the HuggingFace Hub, you must specify the `option.model_id=<hf_hub_model_id>` configuration.
+When using a model directly from the hub, we recommend you also specify the model revision (commit hash) via `option.revision=<commit hash>`.
+Since model artifacts are downloaded at runtime from the Hub, using a specific revision ensures you are using a model compatible with package versions in the runtime environment.
+Open Source model artifacts on the hub are subject to change at any time, and these changes may cause issues when instantiating the model (the model may require a newer version of transformers than what is available in the container). 
+If a model provides custom model (*modeling.py) and custom tokenizer (*tokenizer.py) files, you need to specify `option.trust_remote_code=true` to load and use the model.
+
+If you are deploying a model hosted in S3, `option.model_id=<s3 uri>` should be the s3 object prefix of the model artifacts.
+Alternatively, you can deploy the model artifacts alongside your `serving.properties` configuration file via the `model.tar.gz` artifact and not specify `option.model_id`.
+SageMaker will extract the contents of the `model.tar.gz` archive to the default directory LMI uses to load and serve models.
+
+## Inference Library Configuration
+
+LMI expects the following two configurations to determine which backend to use:
+
+* `engine`. The options are `Python` and `MPI`, which dictates how we launch the Python processes
+* `option.rolling_batch`. The available options depend on which container you use:
+  * The DeepSpeed Container provides `auto`, `deepspeed`, `lmi-dist`, `vllm`
+  * The TensorRT-LLM Container provides `auto`, `tensorrtllm`
+  * The Transformers NeuronX Container provides `auto`
+
+We recommend using `engine=MPI` and `option.rolling_batch=auto` unless you want to use a specific backend.
```

Comment:
MPI + vLLM will not work, we need to make sure we tell the engines of choice when they choose engine

Reply:
I'll provide a static mapping here instead for the supported combinations